dbf-reader/src/main/java/org/jamel/dbf/structure/DbfHeader.java

```diff
@@ -109,18 +107,17 @@ public short getRecordLength() {
         return recordLength;
     }
 
-    int getFieldIndex(String fieldName) {
+    public int getFieldIndex(String fieldName) {
         if (fieldIndexesByNames == null) {
             initFieldIndexesByNames();
         }
-        if (!fieldIndexesByNames.containsKey(fieldName)) {
-            throw new DbfException(format("Field \"%s\" does not exist", fieldName));
-        }
-        return fieldIndexesByNames.get(fieldName);
+        Integer index = fieldIndexesByNames.get(fieldName);
+        return index == null ? -1 : index;
     }
 
     private void initFieldIndexesByNames() {
-        fieldIndexesByNames = new HashMap<>();
+        int capacity = (int) (fields.size() / 0.75 + 1);
+        fieldIndexesByNames = new HashMap<>(capacity);
```

Comment:
It is true for JDK < 1.7u40. See commit http://hg.openjdk.java.net/jdk7u/jdk7u/jdk/diff/99b3f9c060b9/src/share/classes/java/util/HashMap.java into OpenJDK source code. First time `loadFactor` is not used to calculate `threshold`.

But anyway. Code looks strange for me because you are using here some magic constants. It is cleaner to use `HashMap(int initialCapacity, float loadFactor)` with `loadFactor = 1.0f` and not speculate about internal default load factor.